### DISCUSSION TITLE
Add a webhook for each stack

### DISCUSSION THREAD

**Discussion by jtenniswood** [louislam/dockge#178] [Feature Request]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

UI Feature

### 🔖 Feature description

If each stack supported restarting using a webhook, this would help with home automations (such as home assistant)

### ✔️ Solution

Add webhook support

**Comment by watzon** [comment]
This would be so nice! Portainer offers webhook support, but it's locked behind the business edition.

**Comment by cetuuus** [comment]
hey @jtenniswood, are you working on it? I'd like the same thing as you, and I'm ready to try adding this feature.

**Reply by jtenniswood** [reply]
No I don't have the skill to do this, but would still love to use it.

**Reply by watzon** [reply]
Can't wait to see what you come up with @dev-cetus. This is the one thing I'm waiting for before I would consider Dockge perfect for my needs.

**Comment by DanielTezolin** [comment]
This feature of updating the stack via webhook would be perfect for implementing a CI/CD. I use portainer business, but the license of over 1000 dollars for a company that earns in reais is out of touch with reality. 
@louislam, I would like to know if this feature is already in the plans, and if not, I am willing to create a pull request for it.

**Comment by tippfehlr** [comment]
I’ve made [composehook](https://github.com/tippfehlr/composehook) to create webhooks from docker compose labels.
I agree that webhooks in dockge would be :fire: but this could be a substitution until then.

**Reply by cetuuus** [reply]
I've tried to understand the code to implement webhooks, it's a real mess. I think your solution is the best.

**Comment by LeVoMihalcea** [comment]
@louislam anything scheduled on this one? Can the community start looking into this? 😇

**Comment by jadenzaleski** [comment]
I would also like this feature to be added. It would be very useful for CI/CD pipelines.

**Comment by Aerya** [comment]
Hi, added to my Enhanced fork.
